service/accounts/openstack_manager.py

```diff
@@ -230,11 +232,16 @@ def create_account(self, username, password=None, project_name=None,
 
         try:
             allocation_source_name = username
-            # Create Allocation Source for User
-            self._create_allocation_source(allocation_source_name)
-            # Add User to Allocation Source
-            self._assign_user_allocation_source(allocation_source_name,username)
-
+            # Check if allocation source exists
+            source = AllocationSource.objects.filter(name=username)
+            if not source:
+                logger.debug('No Allocation Source with name %s found. Creating a new Allocation Source...', allocation_source_name)
+                # Create Allocation Source for User
+                self._create_allocation_source(allocation_source_name)
+                # Add User to Allocation Source
+                self._assign_user_allocation_source(allocation_source_name,username)
+            else:
+                logger.debug('Allocation Source with name %s exists. Skipping creation step...' , allocation_source_name)
```

Comment:
In the next revision, lets go ahead and encapsulate this as `create_allocation_source_for_user`.

Bonus points: Make a `AllocationSourcePluginManager` that handles this logic, so that _other organizations_ can do something different..